rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -1189,10 +1189,24 @@ static T ppu_load_acquire_reservation(ppu_thread& ppu, u32 addr)
 		ppu.use_full_rdata = false;
 	}
 
-	for (u64 count = 0;; [&]()
+	for (u64 count = 0; count != umax; [&]()
 	{
 		if (ppu.state)
 		{
+			if (ppu.state & cpu_flag::pause)
+			{
+				verify(HERE), cpu_thread::if_suspended<-1>(&ppu, [&]()
+				{
+					// Guaranteed success
+					ppu.rtime = vm::reservation_acquire(addr, sizeof(T)) & -128;
+					mov_rdata(ppu.rdata, *vm::get_super_ptr<spu_rdata_t>(addr & -128));
+				});
+
+				// Exit loop
+				count = -1;
+				return;
```

Comment:
Its a lambda, return does not function as it seems here.

Reply:
Well, I set count to umax to trigger loop exit condition.